test/unit/core/metafunction/range_iterator_test.cpp

```diff
@@ -0,0 +1,278 @@
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#include <list>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <meta/meta.hpp>
+
+#include <range/v3/view/iota.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+
+#include <seqan3/core/metafunction/all.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+using namespace seqan3;
+
+// TODO for most tests: also add something with proxy reference type
+
+TEST(range_and_iterator, iterator_)
+{
+    EXPECT_TRUE((std::is_same_v<iterator_t<std::vector<int>>,
+                                typename std::vector<int>::iterator>));
+    EXPECT_TRUE((std::is_same_v<iterator_t<std::vector<int> const>,
+                                typename std::vector<int>::const_iterator>));
+
+    auto v = ranges::view::ints(1);
+    EXPECT_TRUE((std::is_same_v<iterator_t<decltype(v)>,
+                                decltype(ranges::begin(v))>));
+    EXPECT_FALSE((std::is_same_v<iterator_t<decltype(v)>,
+                                 decltype(ranges::end(v))>));
+}
+
+TEST(range_and_iterator, sentinel_)
+{
+    EXPECT_TRUE((std::is_same_v<sentinel_t<std::vector<int>>,
+                                typename std::vector<int>::iterator>));
+    EXPECT_TRUE((std::is_same_v<sentinel_t<std::vector<int> const>,
+                                typename std::vector<int>::const_iterator>));
+    EXPECT_TRUE((std::is_same_v<sentinel_t<std::vector<int>>,
+                                iterator_t<std::vector<int>>>));
+
+    auto v = ranges::view::ints(1);
+    EXPECT_FALSE((std::is_same_v<sentinel_t<decltype(v)>,
+                                decltype(ranges::begin(v))>));
+    EXPECT_TRUE((std::is_same_v<sentinel_t<decltype(v)>,
+                                decltype(ranges::end(v))>));
+}
+
+template <typename list1, typename list2, size_t pos = 0>
+void assert_same_types()
+{
+    EXPECT_TRUE((std::is_same_v<meta::at_c<list1, pos>, meta::at_c<list2, pos>>));
+    if constexpr (pos < list1::size() - 1)
+        assert_same_types<list1, list2, pos + 1>();
```

Comment:
I think you overuse `meta::` in situations where you can use parameter packs.

and do something like this
```c++
template <typename ...ts1, typename ...ts2>
bool same_zipped_types(std::tuple<ts1>, std::tuple<ts2>)
{
     return (std::is_same_v<ts1, ts2> && ... && true);
}
```

Reply:
not viable, as discussed